svgnative/src/SVGNativeCWrapper.cpp

```diff
@@ -239,20 +239,30 @@ void svg_native_render_size(svg_native_t* sn, float width, float height)
 }
 
 #ifdef USE_TEXT
-void svg_native_render_output(svg_native_t* sn, char* output, int& length)
+void svg_native_get_output(svg_native_t* sn, const char** data, size_t* length)
 {
+    if (!length || !data)
+    {
+        return;
+    }
+
+    *data = NULL;
+    *length = 0;
+
     auto _sn = dynamic_cast<svg_native_t_*>(sn);
     if (!_sn || _sn->mRendererType != SVG_RENDERER_STRING)
         return;
 
     auto renderer = std::dynamic_pointer_cast<SVGNative::StringSVGRenderer>(_sn->mRenderer);
     if (!renderer)
         return;
+
     const auto& string = renderer->String();
-    auto outputLength = std::min(length, static_cast<int>(string.size()));
-    length = outputLength;
-    for (int i = 0; i < outputLength; i++)
-        output[i] = string[i]; 
+
+    *data = string.c_str();
+    *length = string.size();
```

Comment:
As written in another comment, you need to copy the string. You can not just pass the pointer to the string since the string gets cleaned up when the renderer gets cleaned up. Maybe we should still limit the max string size to a certain amount. Could be more than the 2048 it was before though.

Reply:
Umm, at present, my draft is something like this:
```
#ifdef USE_TEXT
void svg_native_get_output(svg_native_t* sn, char** buff, size_t* length)
{
    if (!length)
        return;

    auto _sn = dynamic_cast<svg_native_t_*>(sn);
    if (!_sn || _sn->mRendererType != SVG_RENDERER_STRING)
        return;

    auto renderer = std::dynamic_pointer_cast<SVGNative::StringSVGRenderer>(_sn->mRenderer);
    if (!renderer)
        return;

    const auto& string = renderer->String();

    /*
     * a) buff == NULL: return the length only with no data.
     * b) *buff != NULL: copy the data to preallocated buffer. *length is updated.
     * c) *buff == NULL: allocate new buffer to copy the data.
     */

    /* a) no copy, return the length. */
    if (!buff)
    {
        *length = string.size() + 1;
        return;
    }

    /* b) copy to preallocated buffer */
    if (*buff)
    {
        if (!(*length))
            return;

        memset(buff, 0, *length);
        if (string.size() < *length)
            *length = string.size();
        memcpy(buff, string.c_str(), *length);
        return;
    }

    /* c) copy to new buffer */
    *length = string.size();
    *buff = (char*)malloc(*length);
    memcpy(*buff, string.c_str(), *length);

    return;
}
#endif
```
The reason of complexity is "how to retrieve whole data, and how to deal the buffer allocated by the client (of this library)". I understand this API is not for the real usage but the sample, so I can reduce some features. How do you think of?

Reply:
I think we should:
* set `*buff` to `NULL` and `*length` to `0`,
* then we use the `auto size = string.length() + 1;` (should we still limit the max string length? Maybe not.)
* then allocate the buffer `*buf = (const char*) malloc(size * sizeof(const char*));`
* `memcpy(*buff, string.c_str(), size);` (and set the null termination at the end manually or use memset first as you did before.)
* Last but not least set `*length` to `size` and return.

No need to check if the caller created some buffer before. The caller is not supposed to handle this.

Reply:
Note, if we do not limit the string length manually, there is no need to use memset or manually set the last char to null terminal. Probably not worth it to keep the limitation for the purpose of demonstrating the interface.

Reply:
Oh, sorry my commit is too late and made a conflict with the symbol export patch. I would do rebase.